container/Containerfile

```diff
@@ -0,0 +1,233 @@
+    if [ "${CEPH_VERSION_NAME}" == "main" ] || ${IS_DEVEL}; then \
+        ARCH=$(arch); if [ "${ARCH}" == "aarch64" ]; then ARCH="arm64"; fi ;\
+        REPO_URL=$(curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/8/${ARCH}&flavor=${OSD_FLAVOR}&ref=${CEPH_REF}&sha1=latest" | jq -r .[0].url) ;\
+        IS_RELEASE=0 ;\
+        echo "" > crimson-packages.txt ;\
+        if [ "${OSD_FLAVOR}" == "crimson" ]; then \
+            echo "ceph-crimson-osd${CEPH_PACKAGE_VERSION}" > crimson-packages.txt ;\
+        fi ;\
+    else \
+        IS_RELEASE=1 ;\
+        REPO_URL="http://download.ceph.com/rpm-${CEPH_VERSION_NAME}/el8/" ;\
+        echo "" > crimson-packages.txt ;\
+    fi && \
+    rpm -Uvh "$REPO_URL/noarch/ceph-release-1-${IS_RELEASE}.el8.noarch.rpm"
+
+# Copr repos
+# TODO: afaict, scikit is for mgr-diskpredition-local, asyncssh for cephadm, but why do we
+#       encode these as deps only in container instead of including them as a vendored
+#       submodule in the ceph repo or something to be included with ceph packages?
+#       @tchaikov @torchiaf @adk3798 @ktdreyer?
+#       ref: https://github.com/ceph/ceph-container/pull/1821
+#       note: scikit-learn packages are not included in ubi8 in ceph-container
+RUN \
+    dnf install -y --setopt=install_weak_deps=False dnf-plugins-core && \
+    dnf copr enable -y tchaikov/python-scikit-learn && \
+    dnf copr enable -y tchaikov/python3-asyncssh
+
+# Update package mgr
+RUN dnf update -y --setopt=install_weak_deps=False
+
+# Define and install packages
+# TODO: review these packages
+RUN \
+    # General
+    PACKAGES="ca-certificates" && \
+    # Ceph
+    # TODO: I removed these packages that were included in ganesha packages:
+    #       gcc, lua-devel, luarocks
+    #    They are listed as BuildRequires in ceph.spec.in, so I assume they are unnecessary
+    #    @mattbenjamin please advise about whether these are necessary for runtime
+    PACKAGES="$PACKAGES \
+        ceph-common${CEPH_PACKAGE_VERSION} \
+        ceph-exporter${CEPH_PACKAGE_VERSION} \
+        ceph-grafana-dashboards${CEPH_PACKAGE_VERSION} \
+        ceph-immutable-object-cache${CEPH_PACKAGE_VERSION} \
+        ceph-mds${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-cephadm${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-dashboard${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-diskprediction-local${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-k8sevents${CEPH_PACKAGE_VERSION} \
+        ceph-mgr-rook${CEPH_PACKAGE_VERSION} \
+        ceph-mgr${CEPH_PACKAGE_VERSION} \
+        ceph-mon${CEPH_PACKAGE_VERSION} \
+        ceph-osd${CEPH_PACKAGE_VERSION} \
+        ceph-radosgw${CEPH_PACKAGE_VERSION} \
+        ceph-volume${CEPH_PACKAGE_VERSION} \
+        cephfs-mirror${CEPH_PACKAGE_VERSION} \
+        cephfs-top${CEPH_PACKAGE_VERSION} \
+        kmod \
+        libradosstriper1${CEPH_PACKAGE_VERSION} \
+        rbd-mirror${CEPH_PACKAGE_VERSION} \
+        $(cat crimson-packages.txt) \
```

Comment:
@mattbenjamin could you advise if gcc, lua-devel, or luarocks packages are needed in the ceph container image?

Reply:
lua-devel and luarocks are runtime dependencies for radosgw, and https://github.com/ceph/ceph/pull/52931 (cc @yuvalif) is moving them from BuildRequires to Requires

Reply:
Added those back in here. Left `gcc` removed for now.